### PR TITLE
New gamemode, Civil War

### PR DESCRIPTION
**Pull request by BraveMole**
## About The Pull Request

This is a first batch of the HvH mode. It is playable-ish, it still lacks things like objectives or even a balance function.

WARNING: this is only playable on the ship map = twin pillars and on the ground map ice colony

This adds to the other PR:

The twin pillars shipmap, it's a map with two identical ships, one for the rebel faction, the other for the loyalist faction
Some small refactoring of the join system, so you are able to join the loyalist or rebel faction. This is very much WIP and will be improved in later PR, but this is good enough for some test
Another shuttle, the triumph. It's a copy of the alamo, but for the rebels.
Add the energy weapon in HvH vendors since david forgot to do it

This is meant for one or two tests rounds with probably a lot of admin intervention, as a lot is to be tested.

## Why It's Good For The Game

New gamemode

## Changelog
:cl:
add: New HvH gamemode, civil war. Very much WIP
/:cl:

### CODE REVIEW

**Comment by DerrickChamberlain** [comment]
Exciting

**Review by TiviPlus — APPROVED** [review]
code ok, why are triumph and victory both alamo clones

File: code/datums/gamemodes/game_mode.dm [hunk 1]
```diff
@@ -668,21 +680,24 @@ Sensors indicate [numXenosShip || "no"] unknown lifeform signature[numXenosShip
 	SSjob.set_active_joinable_occupations_by_category()
 	return TRUE
 
+///If joining the job.faction will make the game too unbalanced, return FALSE
+/datum/game_mode/proc/is_faction_balanced(datum/job/job)
+	return TRUE
```
Comment (BraveMole) [inline code/datums/gamemodes/game_mode.dm:685]: WIP

**Comment by JackTheJackhammer** [comment]
Oh Nice!

Question: what do we do about CAS? Will there be some AAA thing that forces CAS away like the AAA in Rising Storm 2 (Its a button for commanders that shoots down the "Spooky" Air support gunship thingy)

**Comment by BraveMole** [comment]
> Oh Nice!
> 
> Question: what do we do about CAS? Will there be some AAA thing that forces CAS away like the AAA in Rising Storm 2 (Its a button for commanders that shoots down the "Spooky" Air support gunship thingy)

No cas on the ship, no OB either

**Pull request closed without merging**